doc/devdocs/types.rst

```diff
@@ -0,0 +1,522 @@
+****************
+More about types
+****************
+
+.. currentmodule:: Base
+
+If you've used Julia for a while, you understand the fundamental role
+that types play.  Here we try to get under the hood, focusing
+particularly on :ref:`parametric types <man-parametric-types>`.
+
+Types and sets (and ``Any`` and ``Union()``/``Bottom``)
+-------------------------------------------------------
+
+It's perhaps easiest to conceive of Julia's type system in terms of
```

Comment:
I personally feel like this sentence is the best one I've seen that explains the connection between different pieces. =)